# jEETENGE TO HUM HI by gaurav

### Load libraries and data

In [1]:
import pandas as pd
import numpy as np
# import lightgbm as lgb
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder

In [2]:
train = pd.read_csv("train.csv")
test = pd.read_csv("test.csv")

In [3]:
print("5train shape", train.shape)
print("test shape", test.shape)

5train shape (12137810, 10)
test shape (3706907, 9)


In [3]:
# check missing values per column
train.isnull().sum(axis=0)/train.shape[0]

ID             0.000000
datetime       0.000000
siteid         0.099896
offerid        0.000000
category       0.000000
merchant       0.000000
countrycode    0.000000
browserid      0.050118
devid          0.149969
click          0.000000
dtype: float64

In [4]:
train.click.value_counts()/train.shape[0]

0    0.963979
1    0.036021
Name: click, dtype: float64

In [5]:
train[train.click==1].shape

(437214, 10)

### Clean Data and Create Features

In [6]:
train['siteid'].fillna(-999, inplace=True)
test['siteid'].fillna(-999, inplace=True)

train['browserid'].fillna("None", inplace=True)
test['browserid'].fillna("None", inplace=True)

train['devid'].fillna("None", inplace=True)
test['devid'].fillna("None", inplace=True)

In [7]:
# set datatime
train['datetime'] = pd.to_datetime(train['datetime'])
test['datetime'] = pd.to_datetime(test['datetime'])

In [8]:
# create datetime variable
train['tweekday'] = train['datetime'].dt.weekday
train['thour'] = train['datetime'].dt.hour
train['tminute'] = train['datetime'].dt.minute

test['tweekday'] = test['datetime'].dt.weekday
test['thour'] = test['datetime'].dt.hour
test['tminute'] = test['datetime'].dt.minute

In [9]:
cols = ['siteid','offerid','category','merchant']

for x in cols:
    train[x] = train[x].astype('object')
    test[x] = test[x].astype('object')

In [10]:
cat_cols = cols + ['countrycode','browserid','devid']

In [11]:
for col in cat_cols:
    lbl = LabelEncoder()
    lbl.fit(list(train[col].values) + list(test[col].values))
    train[col] = lbl.transform(list(train[col].values))
    test[col] = lbl.transform(list(test[col].values))

In [12]:
train.head()

ID            datetime  siteid  offerid  category  merchant  \
0  IDsrk7SoW 2017-01-14 09:42:09  128865   784773        48       127   
1  IDmMSxHur 2017-01-18 17:50:53  142053   157563        59        65   
2  IDVLNN0Ut 2017-01-11 12:46:49    2618   458279        69        15   
3  ID32T6wwQ 2017-01-17 10:18:43  243406   345067       117       507   
4  IDqUShzMg 2017-01-14 16:02:33  154278   417948        36       276   

   countrycode  browserid  devid  click  tweekday  thour  tminute  
0            4          2      2      0         5      9       42  
1            1          8      0      0         2     17       50  
2            0          1      2      0         2     12       46  
3            2          2      1      0         1     10       18  
4            3          8      0      0         5     16        2

### Model Training

In [13]:
cols_to_use = list(set(train.columns) - set(['ID','datetime','click']))

In [14]:
from sklearn.model_selection import StratifiedKFold
skf = StratifiedKFold(n_splits=20)
a = 0
for i, j in skf.split(train[cols_to_use], train['click']):
    a = j
len(a)

606889

In [15]:
X_train, X_test, y_train, y_test = train_test_split(train.loc[a][cols_to_use], train.loc[a]['click'], test_size = 0.5)

In [ ]:
y_test.value_counts()

0    292444
1     11001
Name: click, dtype: int64

In [ ]:
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import roc_auc_score
import time
t1 = time.time()
dict1 = {}
dict2 = {}
for i in range(300, 600, 10):
    
    print("Model with ", i, " n estimators")
    model_mod = RandomForestClassifier(n_estimators=i, verbose=1, oob_score=True, n_jobs=-1, random_state=42)#lets use the fucing cores
    start = time.time()
    model_mod.fit(X_train, y_train)
    print("time to fit: ", time.time()-start)
    start = time.time()
    pred = model_mod.predict(X_test)
    print("time to predict: ", time.time()-start ,"\n")
    print("training score: ", roc_auc_score(y_train, model_mod.predict(X_train)))
    print("crossval score: ", roc_auc_score(y_test, pred))
    print("oob score: ", model_mod.oob_score_)
    dict1[i]=roc_auc_score(y_test, pred)
    dict2[i]=model_mod.oob_score_
    print("Done :: ", i)
print("total time ", time.time()-t1)

Model with  300  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.4s
[Parallel(n_jobs=-1)]: Done 300 out of 300 | elapsed:   29.2s finished


time to fit:  39.16286563873291


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 300 out of 300 | elapsed:    1.8s finished


time to predict:  2.3157787322998047 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 300 out of 300 | elapsed:    1.7s finished


training score:  1.0
crossval score:  0.797898690725
oob score:  0.97775207287
Done ::  300
Model with  310  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.3s
[Parallel(n_jobs=-1)]: Done 310 out of 310 | elapsed:   29.6s finished


time to fit:  40.13161849975586


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 310 out of 310 | elapsed:    1.9s finished


time to predict:  2.343656539916992 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 310 out of 310 | elapsed:    1.9s finished


training score:  1.0
crossval score:  0.798117394148
oob score:  0.977725708862
Done ::  310
Model with  320  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.3s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.9s
[Parallel(n_jobs=-1)]: Done 320 out of 320 | elapsed:   31.8s finished


time to fit:  42.57401657104492


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 320 out of 320 | elapsed:    1.9s finished


time to predict:  2.4508986473083496 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 320 out of 320 | elapsed:    1.9s finished


training score:  1.0
crossval score:  0.798258874576
oob score:  0.977725708862
Done ::  320
Model with  330  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.4s
[Parallel(n_jobs=-1)]: Done 330 out of 330 | elapsed:   32.2s finished


time to fit:  43.37625980377197


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 330 out of 330 | elapsed:    2.0s finished


time to predict:  2.497302532196045 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 330 out of 330 | elapsed:    2.0s finished


training score:  1.0
crossval score:  0.798162844562
oob score:  0.977788323381
Done ::  330
Model with  340  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.3s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.9s
[Parallel(n_jobs=-1)]: Done 340 out of 340 | elapsed:   33.9s finished


time to fit:  45.55241060256958


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.1s
[Parallel(n_jobs=12)]: Done 340 out of 340 | elapsed:    2.1s finished


time to predict:  2.632237195968628 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 340 out of 340 | elapsed:    2.1s finished


training score:  1.0
crossval score:  0.798171393207
oob score:  0.977725708862
Done ::  340
Model with  350  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.8s
[Parallel(n_jobs=-1)]: Done 350 out of 350 | elapsed:   34.6s finished


time to fit:  46.47461223602295


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 350 out of 350 | elapsed:    2.1s finished


time to predict:  2.716000556945801 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 350 out of 350 | elapsed:    2.1s finished


training score:  1.0
crossval score:  0.798526447871
oob score:  0.977725708862
Done ::  350
Model with  360  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.8s
[Parallel(n_jobs=-1)]: Done 360 out of 360 | elapsed:   35.5s finished


time to fit:  47.702343225479126


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.1s
[Parallel(n_jobs=12)]: Done 360 out of 360 | elapsed:    2.2s finished


time to predict:  2.7589638233184814 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 360 out of 360 | elapsed:    2.2s finished


training score:  1.0
crossval score:  0.798297486073
oob score:  0.977745481868
Done ::  360
Model with  370  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.7s
[Parallel(n_jobs=-1)]: Done 370 out of 370 | elapsed:   35.9s finished


time to fit:  48.15026521682739


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.1s
[Parallel(n_jobs=12)]: Done 370 out of 370 | elapsed:    2.3s finished


time to predict:  2.835028648376465 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 370 out of 370 | elapsed:    2.2s finished


training score:  1.0
crossval score:  0.798420159482
oob score:  0.977745481868
Done ::  370
Model with  380  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   18.1s
[Parallel(n_jobs=-1)]: Done 380 out of 380 | elapsed:   37.6s finished


time to fit:  50.34612464904785


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 380 out of 380 | elapsed:    2.3s finished


time to predict:  2.902085304260254 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.1s
[Parallel(n_jobs=12)]: Done 380 out of 380 | elapsed:    2.4s finished


training score:  1.0
crossval score:  0.798372999339
oob score:  0.977758663872
Done ::  380
Model with  390  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.5s
[Parallel(n_jobs=-1)]: Done 390 out of 390 | elapsed:   37.7s finished


time to fit:  50.601012229919434


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 390 out of 390 | elapsed:    2.3s finished


time to predict:  2.840020179748535 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 390 out of 390 | elapsed:    2.3s finished


training score:  1.0
crossval score:  0.798502511664
oob score:  0.977788323381
Done ::  390
Model with  400  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.6s
[Parallel(n_jobs=-1)]: Done 400 out of 400 | elapsed:   39.4s finished


time to fit:  53.21878242492676


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 400 out of 400 | elapsed:    2.4s finished


time to predict:  3.020939350128174 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 400 out of 400 | elapsed:    2.3s finished


training score:  1.0
crossval score:  0.798869534431
oob score:  0.977765254874
Done ::  400
Model with  410  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.3s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.4s
[Parallel(n_jobs=-1)]: Done 410 out of 410 | elapsed:   39.6s finished


time to fit:  53.29703402519226


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 410 out of 410 | elapsed:    2.4s finished


time to predict:  2.961107015609741 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 410 out of 410 | elapsed:    2.4s finished


training score:  1.0
crossval score:  0.7989570158
oob score:  0.977725708862
Done ::  410
Model with  420  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.4s
[Parallel(n_jobs=-1)]: Done 420 out of 420 | elapsed:   40.7s finished


time to fit:  54.76736497879028


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 420 out of 420 | elapsed:    2.5s finished


time to predict:  3.1271989345550537 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 420 out of 420 | elapsed:    2.5s finished


training score:  1.0
crossval score:  0.798729763732
oob score:  0.977765254874
Done ::  420
Model with  430  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.5s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   18.4s
[Parallel(n_jobs=-1)]: Done 430 out of 430 | elapsed:   43.0s finished


time to fit:  57.46650981903076


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 430 out of 430 | elapsed:    2.6s finished


time to predict:  3.2173120975494385 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 430 out of 430 | elapsed:    2.5s finished


training score:  1.0
crossval score:  0.798997337026
oob score:  0.977788323381
Done ::  430
Model with  440  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.1s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.3s
[Parallel(n_jobs=-1)]: Done 440 out of 440 | elapsed:   42.7s finished


time to fit:  57.36681771278381


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 440 out of 440 | elapsed:    2.6s finished


time to predict:  3.285334587097168 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 440 out of 440 | elapsed:    2.6s finished


training score:  1.0
crossval score:  0.799456970349
oob score:  0.977808096387
Done ::  440
Model with  450  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.4s
[Parallel(n_jobs=-1)]: Done 426 tasks      | elapsed:   41.7s
[Parallel(n_jobs=-1)]: Done 450 out of 450 | elapsed:   43.6s finished


time to fit:  58.45259642601013


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.6s
[Parallel(n_jobs=12)]: Done 450 out of 450 | elapsed:    2.7s finished


time to predict:  3.301347255706787 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 450 out of 450 | elapsed:    2.6s finished


training score:  1.0
crossval score:  0.799418358852
oob score:  0.977798209884
Done ::  450
Model with  460  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.1s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.4s
[Parallel(n_jobs=-1)]: Done 426 tasks      | elapsed:   42.5s
[Parallel(n_jobs=-1)]: Done 460 out of 460 | elapsed:   45.3s finished


time to fit:  60.725345611572266


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.6s
[Parallel(n_jobs=12)]: Done 460 out of 460 | elapsed:    2.8s finished


time to predict:  3.4004225730895996 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 460 out of 460 | elapsed:    2.7s finished


training score:  1.0
crossval score:  0.799497291576
oob score:  0.97781798289
Done ::  460
Model with  470  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.5s
[Parallel(n_jobs=-1)]: Done 426 tasks      | elapsed:   41.8s
[Parallel(n_jobs=-1)]: Done 470 out of 470 | elapsed:   45.7s finished


time to fit:  61.4837532043457


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 470 out of 470 | elapsed:    2.7s finished


time to predict:  3.441448211669922 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 470 out of 470 | elapsed:    2.8s finished


training score:  1.0
crossval score:  0.799123429893
oob score:  0.977811391888
Done ::  470
Model with  480  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.4s
[Parallel(n_jobs=-1)]: Done 426 tasks      | elapsed:   41.6s
[Parallel(n_jobs=-1)]: Done 480 out of 480 | elapsed:   46.4s finished


time to fit:  62.20841145515442


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 480 out of 480 | elapsed:    2.8s finished


time to predict:  3.534534215927124 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 480 out of 480 | elapsed:    2.9s finished


training score:  1.0
crossval score:  0.798948467155
oob score:  0.977788323381
Done ::  480
Model with  490  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.2s
[Parallel(n_jobs=-1)]: Done 426 tasks      | elapsed:   41.5s
[Parallel(n_jobs=-1)]: Done 490 out of 490 | elapsed:   47.2s finished


time to fit:  63.214982748031616


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.1s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.6s
[Parallel(n_jobs=12)]: Done 490 out of 490 | elapsed:    3.0s finished


time to predict:  3.711703062057495 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.6s
[Parallel(n_jobs=12)]: Done 490 out of 490 | elapsed:    2.9s finished


training score:  1.0
crossval score:  0.798670635486
oob score:  0.977831164894
Done ::  490
Model with  500  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.1s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.4s
[Parallel(n_jobs=-1)]: Done 426 tasks      | elapsed:   41.5s
[Parallel(n_jobs=-1)]: Done 500 out of 500 | elapsed:   48.3s finished


time to fit:  64.61894798278809


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 500 out of 500 | elapsed:    3.0s finished


time to predict:  3.7677080631256104 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 500 out of 500 | elapsed:    3.0s finished


training score:  1.0
crossval score:  0.798939918509
oob score:  0.977864119904
Done ::  500
Model with  510  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.4s
[Parallel(n_jobs=-1)]: Done 426 tasks      | elapsed:   41.7s
[Parallel(n_jobs=-1)]: Done 510 out of 510 | elapsed:   49.3s finished


time to fit:  66.08902668952942


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 510 out of 510 | elapsed:    3.0s finished


time to predict:  3.706636667251587 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 510 out of 510 | elapsed:    3.1s finished


training score:  1.0
crossval score:  0.799123429893
oob score:  0.977831164894
Done ::  510
Model with  520  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.4s
[Parallel(n_jobs=-1)]: Done 426 tasks      | elapsed:   41.5s
[Parallel(n_jobs=-1)]: Done 520 out of 520 | elapsed:   50.1s finished


time to fit:  67.19981360435486


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.6s
[Parallel(n_jobs=12)]: Done 520 out of 520 | elapsed:    3.1s finished


time to predict:  3.9089341163635254 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 520 out of 520 | elapsed:    3.1s finished


training score:  1.0
crossval score:  0.798862695515
oob score:  0.9778509379
Done ::  520
Model with  530  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.5s
[Parallel(n_jobs=-1)]: Done 426 tasks      | elapsed:   41.6s
[Parallel(n_jobs=-1)]: Done 530 out of 530 | elapsed:   51.2s finished


time to fit:  68.60696744918823


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 530 out of 530 | elapsed:    3.2s finished


time to predict:  3.954882860183716 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 530 out of 530 | elapsed:    3.1s finished


training score:  1.0
crossval score:  0.799130268809
oob score:  0.977847642399
Done ::  530
Model with  540  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.2s
[Parallel(n_jobs=-1)]: Done 426 tasks      | elapsed:   41.2s
[Parallel(n_jobs=-1)]: Done 540 out of 540 | elapsed:   51.7s finished


time to fit:  69.37424874305725


[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.5s
[Parallel(n_jobs=12)]: Done 540 out of 540 | elapsed:    3.3s finished


time to predict:  4.062948942184448 



[Parallel(n_jobs=12)]: Done  26 tasks      | elapsed:    0.1s
[Parallel(n_jobs=12)]: Done 176 tasks      | elapsed:    1.0s
[Parallel(n_jobs=12)]: Done 426 tasks      | elapsed:    2.6s
[Parallel(n_jobs=12)]: Done 540 out of 540 | elapsed:    3.3s finished


training score:  1.0
crossval score:  0.798908145928
oob score:  0.977844346898
Done ::  540
Model with  550  n estimators


[Parallel(n_jobs=-1)]: Done  26 tasks      | elapsed:    3.2s
[Parallel(n_jobs=-1)]: Done 176 tasks      | elapsed:   17.4s
[Parallel(n_jobs=-1)]: Done 426 tasks      | elapsed:   41.4s
[Parallel(n_jobs=-1)]: Done 550 out of 550 | elapsed:   52.9s finished


400, njob=-1
training score:  1.0
crossval score:  0.799803514688
oob score:  0.978233216013

0    294387
1      9058
Name: 0, dtype: int64
0    292421
1     11024
Name: click, dtype: int64

600, njob=-1
worse performance

470, 510 looks good to me

In [ ]:
pred = pd.DataFrame(pred)
print(pred[0].value_counts())
y_test = pd.Series(y_test)
print(y_test.value_counts())

In [ ]:
model.random_state

In [ ]:
model.

In [ ]:
a = test.columns

In [ ]:
pred = model_mod.predict_proba(test[cols_to_use])

In [ ]:
pred = pd.DataFrame(pred)
pred.shape

In [ ]:
model.feature_importances_

In [ ]:
sub = pd.DataFrame({'ID':test['ID'], 'click':pred[1]})
# sub.to_csv('smallData_rf_1.csv', index=False)